arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,332 @@
+            bufferSizeCell.textLabel?.isEnabled = false
+            bufferSizeCell.isUserInteractionEnabled = false
+        } else {
+            bufferSizeCell.textLabel?.isEnabled = true
+            bufferSizeCell.isUserInteractionEnabled = true
+        }
+    }
+    
+    /// Toggles visisbility of the buffer size scale picker.
+    func toggleBufferSizePickerVisibility() {
+        let bufferSizePicker = IndexPath(row: 3, section: 0)
+        let bufferSizeLabel = bufferSizeCell.detailTextLabel
+        tableView.performBatchUpdates({
+            if bufferSizePickerHidden {
+                bufferSizeLabel?.textColor = .accentColor
+                tableView.insertRows(at: [bufferSizePicker], with: .fade)
+                bufferSizePickerHidden = false
+            } else {
+                bufferSizeLabel?.textColor = nil
+                tableView.deleteRows(at: [bufferSizePicker], with: .fade)
+                bufferSizePickerHidden = true
+            }
+        }, completion: nil)
+    }
+    
+    // MARK: UITableViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    
+    /// Show the list of options according to the selected row.
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        switch cell {
+        case statusCell:
+            showStatusOptions()
+        case protectionCell:
+            showProtectionOptions()
+        case bufferSizeCell:
+            tableView.deselectRow(at: indexPath, animated: true)
+            showBufferSizeOptions()
+            if selectedProtection != nil {
+                toggleBufferSizePickerVisibility()
+            } else {
+                return
+            }
+        default:
+            return
+        }
+    }
+        
+    /* UITableViewDataSource */
+    
+    /// Return the number of rows depending on the picker view.
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let numberOfRows = super.tableView(tableView, numberOfRowsInSection: section)
+        if bufferSizePickerHidden {
+            return numberOfRows - 1
+        } else {
+            return numberOfRows
+        }
+    }
+}
+
+// MARK: Pickerview
+
+extension ContingentValuesTableViewController: UIPickerViewDataSource {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return bufferSizes!.count
+    }
+}
+
+extension ContingentValuesTableViewController: UIPickerViewDelegate {
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        if let bufferSizes = bufferSizes {
+            let bufferSizeTitle = String(bufferSizes[row])
+            return bufferSizeTitle
+        }
+        return ""
```

Comment:
```suggestion
```
Since this delegate method returns a optional string, no need to return an empty string.

Reply:
I made the wrong suggestion here 🤥 . It should be `return nil`. No need to fix now, I'll make another suggestion.